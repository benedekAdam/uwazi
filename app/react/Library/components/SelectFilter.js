import PropTypes from 'prop-types';
import React from 'react';

import { MultiSelect, Switcher } from 'app/ReactReduxForms';

const SelectFilter = ({ onChange, model, label, options, prefix, showBoolSwitch }) => (
  <ul className="search__filter is-active">
    <li>
      {label}
      {showBoolSwitch && <Switcher model={`${model}.and`} prefix={prefix} onChange={onChange} />}
    </li>
    <li className="wide">
      <MultiSelect
        model={`${model}.values`}
        prefix={prefix}
        options={options}
        optionsValue="id"
        onChange={onChange}
      />
    </li>
  </ul>
);

SelectFilter.defaultProps = {
  onChange: () => {},
  label: '',
  prefix: '',
  showBoolSwitch: false,
};

SelectFilter.propTypes = {
  model: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  prefix: PropTypes.string,
  showBoolSwitch: PropTypes.bool,
  options: PropTypes.array.isRequired,
  label: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string
  ])
};

export default SelectFilter;
