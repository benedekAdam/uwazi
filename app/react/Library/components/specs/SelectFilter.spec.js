import React from 'react';

import { shallow } from 'enzyme';

import SelectFilter from '../SelectFilter';

describe('SelectFilter', () => {
  it('should render a text filter field with a label and passing the model', () => {
    const props = {
      label: 'label',
      model: 'model',
      options: ['options'],
    };

    const component = shallow(<SelectFilter {...props}/>);
    expect(component).toMatchSnapshot();
  });

  describe('when showBoolSwitch', () => {
    it('should render the and/or bool switch', () => {
      const props = {
        label: 'label',
        model: 'model',
        prefix: 'prefix',
        options: ['options'],
        showBoolSwitch: true
      };

      const component = shallow(<SelectFilter {...props}/>);
      expect(component).toMatchSnapshot();
    });
  });
});
