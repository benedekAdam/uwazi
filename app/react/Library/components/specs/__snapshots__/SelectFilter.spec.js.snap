// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilter should render a text filter field with a label and passing the model 1`] = `
<ul
  className="search__filter is-active"
>
  <li>
    label
  </li>
  <li
    className="wide"
  >
    <MultiSelect
      model="model.values"
      onChange={[Function]}
      options={
        Array [
          "options",
        ]
      }
      optionsValue="id"
      prefix=""
    />
  </li>
</ul>
`;

exports[`SelectFilter when showBoolSwitch should render the and/or bool switch 1`] = `
<ul
  className="search__filter is-active"
>
  <li>
    label
    <Switcher
      model="model.and"
      onChange={[Function]}
      prefix="prefix"
    />
  </li>
  <li
    className="wide"
  >
    <MultiSelect
      model="model.values"
      onChange={[Function]}
      options={
        Array [
          "options",
        ]
      }
      optionsValue="id"
      prefix="prefix"
    />
  </li>
</ul>
`;
